Makefile

```diff
@@ -48,7 +52,7 @@ download: ## Download Go modules
 	go mod download
 
 container: ## Build the Contour container image
-	docker build . -t $(IMAGE):$(VERSION)
+	docker build --build-arg GIT_VERSION=$(VERSION) --build-arg GIT_BRANCH=$(BRANCH) --build-arg GIT_SHA=$(SHA) . -t $(IMAGE):$(VERSION)
```

Comment:
Currently `make install` links with `-s`, but the Dockerfile is intended to link with `-w -s`. It's not clear to be that this works as intended and there are upstream issues. That seem to indicate that `GOFLAGS` doesn't combind flags as intended.

Let's ensure that there is a single, canonical build through `make install`.

In the Makefile:
```

# Note we use `=` here to ensure make doesn't expand the build vars if they are already set, since git sin't in the build container.

# Sets the current Git sha
BUILD_SHA = $(shell git rev-parse --verify HEAD)
# Sets the current branch
BUILD_BRANCH = $(shell git branch --show-current)
# ...
BUILD_VERSION = $(VERSION)

GO_BUILD_VARS = \
	github.com/projectcontour/contour/internal/build.Version=${BUILD_VERSION} \
	github.com/projectcontour/contour/internal/build.Sha=${BUILD_SHA} \
	github.com/projectcontour/contour/internal/build.Branch=${BUILD_BRANCH}

GO_TAGS := -tags "oidc gcp"
GO_LDFLAGS := -s $(patsubst %,-X %, $(GO_BUILD_VARS))

install: ## Build and install the contour binary
	go install -mod=readonly -v -ldflags="$(GO_LDFLAGS)" $(GO_TAGS) $(MODULE)/cmd/contour

```

Then the docker file just calls the Makefile:

```
COPY Makefile Makefile

ARG BUILD_BRANCH
ARG BUILD_SHA
ARG BUILD_VERSION

RUN make install \
	    CGO_ENABLED=0 \
	    GOOS=linux \
	    BUILD_VERSION=${BUILD_VERSION} \
	    BUILD_SHA=${BUILD_SHA} \
	    BUILD_BRANCH=${BUILD_BRANCH}
```

Note that the Makefile and Dockerfile pass different link flags, I believe that `-s -w` is the intended combination, however.